server/application-server/src/main/java/de/tum/cit/aet/helios/environment/EnvironmentController.java

```diff
@@ -77,6 +77,15 @@ public ResponseEntity<?> lockEnvironment(@PathVariable Long id) {
     return ResponseEntity.ok(EnvironmentDto.fromEnvironment(environment.get()));
   }
 
+  @EnforceAtLeastWritePermission
+  @PutMapping("/{id}/extend-lock")
+  public ResponseEntity<?> extendEnvironmentLock(@PathVariable Long id) {
+    Optional<Environment> environment = environmentService.extendEnvironmentLock(id);
+    return environment
+        .map(env -> ResponseEntity.ok(EnvironmentDto.fromEnvironment(env)))
+        .orElse(ResponseEntity.notFound().build());
```

Comment:
issue: this afaik still provides no message to the user, not sure if this can happen though. I actually would remove the optional from the service and return only the object instead as we are not using it anyways